File: backend_python/app/config/env.py
"""
Environment configuration
"""
from pydantic_settings import BaseSettings
from typing import List, Optional


class Settings(BaseSettings):
    """Application settings"""
    
    # Environment (read from NODE_ENV for compatibility)
    node_env: str = "development"
    
    @property
    def environment(self) -> str:
        """Get environment"""
        return self.node_env
    
    # Server
    port: int = 3000
    
    # Database
    database_url: str
    
    # JWT
    jwt_secret: str
    jwt_expires_in: str = "1d"
    
    # CORS
    cors_origin: str = "http://localhost:3000"
    
    # Rate Limiting
    rate_limit_window_ms: int = 900000  # 15 minutes
    rate_limit_max: int = 100
    
    # Frontend URL
    frontend_url: str = "http://localhost"
    
    # Email (SMTP)
    smtp_host: Optional[str] = None
    smtp_port: int = 587
    smtp_user: Optional[str] = None
    smtp_pass: Optional[str] = None
    
    class Config:
        env_file = ".env"
        case_sensitive = False
        env_file_encoding = "utf-8"
        
    @property
    def cors_origins(self) -> List[str]:
        """Parse CORS origins from comma-separated string"""
        return [origin.strip() for origin in self.cors_origin.split(",")]
    
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        
        # Validate required fields
        if not self.database_url:
            raise ValueError("DATABASE_URL is required")
        if not self.jwt_secret:
            raise ValueError("JWT_SECRET is required")
        
        # Validate JWT_SECRET strength in production
        if self.environment == "production" and len(self.jwt_secret) < 32:
            raise ValueError(
                "JWT_SECRET must be at least 32 characters long in production. "
                "Use a strong random string."
            )
    


settings = Settings()

File: backend/src/config/env.ts
import 'dotenv/config';

const required = ['DATABASE_URL', 'JWT_SECRET'];

for (const key of required) {
  if (!process.env[key]) {
    throw new Error(`Missing environment variable ${key}`);
  }
}

const environment = (process.env.NODE_ENV ?? 'development') as 'development' | 'production' | 'test';

// Validate JWT_SECRET strength in production
if (environment === 'production' && process.env.JWT_SECRET!.length < 32) {
  throw new Error(
    'JWT_SECRET must be at least 32 characters long in production. Use a strong random string.'
  );
}

export const env = {
  environment,
  port: Number(process.env.PORT ?? 3000),
  databaseUrl: process.env.DATABASE_URL!,
  jwtSecret: process.env.JWT_SECRET!,
  jwtExpiresIn: process.env.JWT_EXPIRES_IN ?? '1d',
  // CORS configuration
  corsOrigin: process.env.CORS_ORIGIN?.split(',').map((origin) => origin.trim()) ?? ['http://localhost:3000'],
  // Rate limiting
  rateLimitWindowMs: Number(process.env.RATE_LIMIT_WINDOW_MS ?? 15 * 60 * 1000), // 15 minutes
  rateLimitMax: Number(process.env.RATE_LIMIT_MAX ?? 100), // 100 requests per window
  // Frontend URL
  frontendUrl: process.env.FRONTEND_URL ?? 'http://localhost',
};

